packages/buttons/src/utils/hasType.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+/**
+ * This checks that a component is the same type of instance of another component
+ * @param {Object} react component object
+ * @param {Function} component to match
+ * @return {Boolean} if they match or not
+ */
+export default function(a, b) {
```

Comment:
Should we do an `undefined` check here or maybe assign an default value? This would blow up if undefined is provided. That might be something we want through so IDK.